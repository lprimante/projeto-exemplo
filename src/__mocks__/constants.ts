export const useApiMockData = [
    {
        userId: 1,
        id: 1,
        title: 'Kaliteye ho≈ügeldiniz',
        completed: false,
    },
]
export const errorMessage = 'Some error'
