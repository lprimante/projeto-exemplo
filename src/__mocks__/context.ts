jest.mock('../context/useTodosContext', () => ({
    useTodosContext: () => ({
        state: {
            isError: false,
            isLoading: true,
            todos: [],
            seletedTodo: {},
        },
        operations: {
            setIsLoading: jest.fn(),
            setIsError: jest.fn(),
            setTodos: jest.fn(),
            setSelectedTodo: jest.fn(),
        },
    }),
}))
