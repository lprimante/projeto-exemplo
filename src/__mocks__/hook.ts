jest.mock('../hooks/useTodos', () => ({
    useTodos: () => ({
        isError: false,
        isLoading: false,
        todos: [
            {
                userId: 1,
                id: 1,
                title: 'abc',
                completed: false,
            },
        ],
        seletedTodo: undefined,
        getTodos: jest.fn(),
        updateTodoComplete: jest.fn(),
        getTodoById: jest.fn(),
        saveTodo: jest.fn(),
    }),
}))
