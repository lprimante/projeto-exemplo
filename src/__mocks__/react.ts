const mockSetInitialState: never[] = []
const initialState = jest.fn((newState = mockSetInitialState) => newState)

jest.mock('react', () => {
    const react = jest.requireActual('react')
    return {
        ...react,
        useCallback: jest.fn(fn => fn),
        useState: jest.fn(() => [initialState(), mockSetInitialState]),
    }
})
