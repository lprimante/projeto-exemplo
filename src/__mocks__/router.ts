const mockUseNavigate = jest.fn()
const mockUseParams = jest.fn()

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockUseNavigate,
    useParams: () => mockUseParams,
}))
