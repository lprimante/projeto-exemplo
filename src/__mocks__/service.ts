// @ts-nocheck
import { useApiMockData } from './constants'

jest.mock('service/todos', () => ({
    useTodosService: () => ({
        getAllTodosItens: jest.fn(() =>
            Promise.resolve({
                json: () => Promise.resolve(useApiMockData),
            }),
        ),
    }),
}))
