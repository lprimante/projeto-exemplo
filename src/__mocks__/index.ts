import './react'
import './context'
import './service'
