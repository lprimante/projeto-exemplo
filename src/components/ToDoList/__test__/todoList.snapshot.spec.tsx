import { render } from '@testing-library/react'
import { ToDoList } from '../index'
import { useApiMockData } from '../../../__mocks__/constants'

it('Snapshot of ToDoList', () => {
    const tree = render(<ToDoList todos={useApiMockData} handleSelectChange={jest.fn()} handleEdit={jest.fn()} onDelete={jest.fn()} />)
    expect(tree).toMatchSnapshot()
})
