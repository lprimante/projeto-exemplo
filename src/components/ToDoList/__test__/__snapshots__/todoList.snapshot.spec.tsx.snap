// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of ToDoList 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-d6ovhd-MuiListItem-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              id="1"
              name="Kaliteye hoşgeldiniz"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <p
            class="MuiTypography-root MuiTypography-h6 css-l5sb9-MuiTypography-root"
          >
            Kaliteye hoşgeldiniz
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1b7oxvh-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Editar
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-1z0p9bf-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Deletar
          </button>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-d6ovhd-MuiListItem-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            id="1"
            name="Kaliteye hoşgeldiniz"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <p
          class="MuiTypography-root MuiTypography-h6 css-l5sb9-MuiTypography-root"
        >
          Kaliteye hoşgeldiniz
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-1b7oxvh-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Editar
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-1z0p9bf-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Deletar
        </button>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
